pyvista/core/objects.py

```diff
@@ -426,8 +529,17 @@ def repeat(self, flag):
         self.SetRepeat(flag)
 
     @property
-    def n_components(self):
-        """Components in the image (e.g. 3 [or 4] for RGB[A])."""
+    def n_components(self) -> int:
+        """Components in the image (e.g. 3 [or 4] for RGB[A]).
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> texture = examples.download_masonry_texture()
+        >>> texture.n_components
+        3
+
+        """
         image = self.to_image()
         return image.active_scalars.shape[1]
```

Comment:
Does this also work for greyscale images? Can't they have scalar (1d array) scalars? If so, it might be safer to compute something like the size of the scalars array divided by the number of the relevant data (points or cells).

The code in `to_array()` seems to anticipate scalars being 1d.

Reply:
It doesn't, and that led me down quite a rabbit hole. The `Texture` class now supports greyscale textures.